actors/builtin/market/market_actor.go

```diff
@@ -216,29 +213,17 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 			deal.Proposal.Provider = provider
 			deal.Proposal.Client = client
 
-			// Before any operations that check the balance tables for funds, execute all deferred
-			// deal state updates.
-			//
-			// Note: as an optimization, implementations may cache efficient data structures indicating
-			// which of the following set of updates are redundant and can be skipped.
-			amountSlashedTotal = big.Add(amountSlashedTotal, st.updatePendingDealStatesForParty(rt, client))
```

Comment:
similar issue to WithdrawBalance

Reply:
@zixuanzh this is processed in cron as needed, unless i'm missing your concern

Reply:
assuming penalties are not processed immediately (which @anorth suggested we probably should), we are processing slashing at the end of every tipset, the following edge case might happen, in 3 consecutive messages, a miner first gets slashed but its not processed, then it PublishStorageDeals which might lock some available funds that should already be slashed or Withdraw in the withdraw case, and then cron processes the slashed funds which may no longer exist. I think this has to do with the underlying BalanceTable accounting but i have been away from the specifics for a while now.

Reply:
this is no longer an issue because it doesnt try to move funds out of lock table. We should also test the market actor more extensively.